src/pytest_postgresql/factories.py

```diff
@@ -49,6 +62,81 @@ def get_config(request):
     return config
 
 
+class DatabaseJanitor:
+    """Manage database state for specific tasks."""
+
+    def __init__(
+            self: DatabaseJanitorType,
+            user: str,
+            host: str,
+            port: str,
+            db_name: str,
+            version: Union[str, float, Version]
+    ) -> None:
+        """
+        Initialize janitor.
+
+        :param user: postgresql username
+        :param host: postgresql host
+        :param port: postgresql port
+        :param db_name: database name
+        :param version: postgresql version number
+        """
+        self.user = user
+        self.host = host
+        self.port = port
+        self.db_name = db_name
+        if not isinstance(version, Version):
+            self.version = parse_version(str(version))
+        else:
+            self.version = version
+
+    def init(self: DatabaseJanitorType) -> None:
+        """Create database in postgresql."""
+        with self.cursor() as cur:
+            cur.execute('CREATE DATABASE "{}";'.format(self.db_name))
+
+    def drop(self: DatabaseJanitorType) -> None:
+        """Drop database in postgresql."""
+        # We cannot drop the database while there are connections to it, so we
+        # terminate all connections first while not allowing new connections.
+        if self.version >= parse_version('9.2'):
+            pid_column = 'pid'
+        else:
+            pid_column = 'procpid'
+        with self.cursor() as cur:
+            cur.execute(
+                'UPDATE pg_database SET datallowconn=false WHERE datname = %s;',
+                (self.db_name,))
+            cur.execute(
+                'SELECT pg_terminate_backend(pg_stat_activity.{})'
+                'FROM pg_stat_activity '
+                'WHERE pg_stat_activity.datname = %s;'.format(pid_column),
+                (self.db_name,))
+            cur.execute('DROP DATABASE IF EXISTS "{}";'.format(self.db_name))
+
+    @contextmanager
+    def cursor(self: DatabaseJanitorType) -> cursor:
+        """Return postgresql cursor."""
+        conn = psycopg2.connect(user=self.user, host=self.host, port=self.port)
+        conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+        cur = conn.cursor()
+        yield cur
+        cur.close()
```

Comment:
Should we close the cursor on exception?

Reply:
You're thinking about finally? Well... we could employ it. The previous version did throw it on user so there should be no change according to previous behaviour